x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -70,9 +69,11 @@ export class RuleDataClient implements IRuleDataClient {
     };
   }
 
-  getWriter(options: { namespace?: string } = {}): RuleDataWriter {
+  async getWriter(options: { namespace?: string } = {}): Promise<RuleDataWriter> {
     const { namespace } = options;
     const alias = getNamespacedAlias({ alias: this.options.alias, namespace });
+    await this.createOrUpdateWriteTarget({ namespace });
+    await this.rolloverAliasIfNeeded({ namespace });
```

Comment:
I'm a little confused about the order here. At the very least, `createOrUpdateWriteTarget` should probably be renamed to something like `createWriteTargetIfNeeded`, because it looks like the update functionality was removed. I don't think these functions should ever _both_ do something (we don't want to create an index and then immediately roll it over), so I think we're better off with some kind of if/else construct here.

Reply:
I'll add an "alias exists" check inside the rollover function, rename `createOrUpdateWriteTarget` to `createWriteTargetIfNeeded`, and remove the `createWriteTargetIfNeeded` call here.